src/model/result.model.ts

```diff
@@ -1,7 +1,7 @@
 export interface ValidationResult {
   type: string;
   succeeded: boolean;
-  message: string;
+  message: string | string[];
```

Comment:
Validation result is a single message, when we allow to create multiple messages.i the validation means that we can offer several messages (e.g. a validator that tells you that time cannot be past or you need to book five days I'm advance) but always the result will return only one

Reply:
So, I don't know where we are using the message as `string[]` provided by FieldValidationArgs or RecordValidationArgs

Reply:
Let's say we want to create an IBAN validator this validator conjtains

ES7921000813610123456789
(this contains encoded country name, bank code, cc code etc...)
And you don't want to limit the valdiation just to "Invalid IBAN", you want to provide meaningfull messages:

- Invalid country code
- Invalid dc number
- Invalid bank number
- Invalid length

You just want to return a single error message in the validator but with a rich meaninful.